MoreLinq.Test/HasDuplicatesTest.cs

```diff
@@ -0,0 +1,231 @@
+            {
+                "FirstElement",
+                "SecondElement",
+                "ThirdElement"
+            };
+
+            var hasDuplicates = stringArray.HasDuplicates(new DummyStringAlwaysTrueComparer());
+
+            Assert.That(hasDuplicates, Is.True);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_With_Null_Comparer_Then_Default_Comparer_Is_Used_And_False_Is_Returned()
+        {
+            var stringArray = new[]
+            {
+                "FirstElement",
+                "SecondElement",
+                "ThirdElement"
+            };
+
+            var hasDuplicates = stringArray.HasDuplicates(null);
+
+            Assert.That(hasDuplicates, Is.False);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Sequence_With_Custom_Always_False_Comparer_Then_False_Is_Returned()
+        {
+            var stringArray = new[]
+            {
+                "FirstElement",
+                "SecondElement",
+                "ThirdElement"
+            };
+
+            var hasDuplicates = stringArray.HasDuplicates(new DummyStringAlwaysFalseComparer());
+
+            Assert.That(hasDuplicates, Is.False);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Multiple_Duplicates_Sequence_With_Custom_Always_False_Comparer_Then_False_Is_Returned()
+        {
+            var stringArray = new[]
+            {
+                "DUPLICATED_STRING",
+                "DUPLICATED_STRING",
+                "DUPLICATED_STRING"
+            };
+
+            var hasDuplicates = stringArray.HasDuplicates(new DummyStringAlwaysFalseComparer());
+
+            Assert.That(hasDuplicates, Is.False);
+        }
+
+        [Test]
+        public void When_Asking_For_Duplicates_On_Multiple_Duplicates_Sequence_Projection_With_Custom_Always_False_Comparer_Then_False_Is_Returned()
+        {
+            var dummyClasses = new DummyClass[]
+            {
+                new("DUPLICATED_STRING"),
+                new("DUPLICATED_STRING"),
+                new("DUPLICATED_STRING")
+            };
+
+            var hasDuplicates = dummyClasses.HasDuplicates(x => x.ComparableString, new DummyStringAlwaysFalseComparer());
+
+            Assert.That(hasDuplicates, Is.False);
+        }
+
+        sealed class DummyClass
+        {
+            public string ComparableString { get; }
+
+            public DummyClass(string comparableString) => ComparableString = comparableString;
+        }
+
+        sealed class DummyStringAlwaysTrueComparer : IEqualityComparer<string>
+        {
+            public bool Equals(string? x, string? y) => true;
+
+            public int GetHashCode(string obj) => 0;
+        }
+
+        sealed class DummyStringAlwaysFalseComparer : IEqualityComparer<string>
+        {
+            public bool Equals(string? x, string? y) => false;
+
+            public int GetHashCode(string obj) => 0;
+        }
+    }
```

Comment:
Tests with a `null` source are missing.

Reply:
FYI `null` checks are covered for all operators via `NullArgumentTest`, which uses reflection to generate and run tests.

Reply:
I had some of those at the start, but I removed them after noticing that NullArgumentTest was doing all that already